generator/src/main/java/com/scottlogic/deg/generator/decisiontree/DecisionTreeSimplifier.java

```diff
@@ -68,22 +68,19 @@ private ConstraintNode simplifySingleOptionDecisions(ConstraintNode node) {
                 (parentConstraint, decisionNode) -> {
                     ConstraintNode firstOption = decisionNode.getOptions().iterator().next();
                     if (parentConstraint.getAtomicConstraints().stream().anyMatch(firstOption.getAtomicConstraints()::contains)) {
-                        return parentConstraint.removeDecisions(Collections.singletonList(decisionNode));
+                        return parentConstraint.builder().removeDecision(decisionNode).build();
                     } else {
-                        return parentConstraint
+                        return parentConstraint.builder()
                             .addAtomicConstraints(firstOption.getAtomicConstraints())
                             .addDecisions(firstOption.getDecisions())
-                            .removeDecisions(Collections.singletonList(decisionNode));
+                            .removeDecision(decisionNode).build();
                     }
                 },
                 (node1, node2) ->
-                    new ConstraintNode(
-                        Stream
-                            .concat(node1.getAtomicConstraints().stream(), node2.getAtomicConstraints().stream())
-                            .collect(Collectors.toList()),
-                        Stream
-                            .concat(node1.getDecisions().stream(), node2.getDecisions().stream())
-                            .collect(Collectors.toList())
-                    ));
+                    new ConstraintNodeBuilder().addAtomicConstraints(Stream
```

Comment:
this big block is a bit hard to read now.

Reply:
Yup. I've cleaned it up a bit